pkg/eventing/manager.go

```diff
@@ -50,111 +54,93 @@ func NewEventingManager(
-	ecReconciler *ecbackend.Reconciler
+func (em EventingManager) DeployPublisherProxy(ctx context.Context, eventing *v1alpha1.Eventing, backendType v1alpha1.BackendType) (*appsv1.Deployment, error) {
+	// update EC reconciler NATS and public config from the data in the eventing CR
+	if err := em.updateNatsConfig(ctx, eventing); err != nil {
+		return nil, err
+	}
+	em.updatePublisherConfig(eventing)
+	// deployment, err := em.ecReconcilerClient.DeployPublisherProxy(ctx, ecBackendType, em.natsConfig, em.backendConfig)
+	deployment, err := em.applyPublisherProxyDeployment(ctx, eventing, backendType)
+	if err != nil {
+		return nil, err
+	}
+	return deployment, nil
 }
 
-func (e *ECReconcilerEventingClient) CreateOrUpdatePublisherProxy(
+func (em *EventingManager) applyPublisherProxyDeployment(
 	ctx context.Context,
-	backendType ecv1alpha1.BackendType,
-	natsConfig env.NATSConfig,
-	backendConfig env.BackendConfig) (*appsv1.Deployment, error) {
-	e.ecReconciler.SetNatsConfig(natsConfig)
-	e.ecReconciler.SetBackendConfig(backendConfig)
-	return e.ecReconciler.CreateOrUpdatePublisherProxyDeployment(ctx, backendType, false)
-}
+	eventing *v1alpha1.Eventing,
+	backendType v1alpha1.BackendType) (*appsv1.Deployment, error) {
+	var desiredPublisher *appsv1.Deployment
+
+	switch backendType {
+	case v1alpha1.NatsBackendType:
+		desiredPublisher = newNATSPublisherDeployment(eventing.Name, eventing.Namespace, em.natsConfig, em.backendConfig.PublisherConfig)
+	case v1alpha1.EventMeshBackendType:
+		desiredPublisher = newEventMeshPublisherDeployment(eventing.Name, eventing.Namespace, em.backendConfig.PublisherConfig)
+	default:
+		return nil, fmt.Errorf("unknown EventingBackend type %q", backendType)
+	}
 
-func (em EventingManager) CreateOrUpdatePublisherProxy(ctx context.Context, eventing *v1alpha1.Eventing) (*appsv1.Deployment, error) {
-	natsBackend := eventing.GetNATSBackend()
-	var ecBackendType ecv1alpha1.BackendType
-	if natsBackend == nil {
-		return nil, fmt.Errorf("NATs backend is not specified in the eventing CR")
+	if err := setOwnerReference(eventing, desiredPublisher, em.Scheme()); err != nil {
+		return nil, fmt.Errorf("failed to set controller reference: %v", err)
 	}
 
-	ecBackendType, err := convertECBackendType(natsBackend.Type)
+	currentPublisher, err := em.kubeClient.GetDeployment(ctx, eventing.Name, eventing.Namespace)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to get Event Publisher deployment: %v", err)
 	}
 
-	// update EC reconciler NATS and public config from the data in the eventing CR
-	if err := em.updateNatsConfig(ctx, eventing); err != nil {
-		return nil, err
+	if currentPublisher != nil {
+		// preserve only allowed annotations
+		desiredPublisher.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
+		for k, v := range currentPublisher.Spec.Template.ObjectMeta.Annotations {
+			if _, ok := allowedAnnotations[k]; ok {
+				desiredPublisher.Spec.Template.ObjectMeta.Annotations[k] = v
+			}
```

Comment:
Whats the purpose of this check? we dont allow users to add any annotations?

Reply:
This removes the not allowed annotations. Only `allowedAnnotations` are allowed to be set. I didn't come up with this idea, the logic comes from the EC.